devito/ir/iet/visitors.py

```diff
@@ -243,10 +244,10 @@ def _gen_value(self, obj, mode=1, masked=()):
                       if getattr(obj.function, k, False) and v not in masked]
 
         if (obj._mem_stack or obj._mem_constant) and mode == 1:
-            strtype = obj._C_typedata
+            strtype = ccode(obj._C_typedata)
```

Comment:
why necessary?

Reply:
Because now `_C_typedata` is the type, not the name because the name can differ depending on the language, which the code (printer) takes care od